UI/ViewModel/Base/BaseItemViewModel.cs

```diff
@@ -178,6 +193,28 @@ public void Apply()
                 {
                     Update();
                 }
+                catch (FaultException<TodoClient.ForeignKeyConstraintFault> ex)
```

Comment:
if you send the error message in exception you will write less code. You need only one catch
 catch (FaultException ex)
                  {
                      AppendErrors(nameof(HasServiceError), "Record cannot be changed: " + ex.ErrorMessage);
                  }